YakimS/assignment3/WordSearcherTest.java

```diff
@@ -0,0 +1,36 @@
+package com.shaya;
+
+import java.util.HashSet;
+
+
+public class WordSearcherTest {
+
+    @org.junit.Test
+    public void allWordsInGrid()
+    {
+        char[][] charsGrid = new char[][]
+                {
+                        {'A', 'A', 'R'},
+                        {'T', 'C', 'D'},
+                };
+
+        HashSet wordsSet = new HashSet();
+        wordsSet.add("CAR");
+        wordsSet.add("CARD");
+        wordsSet.add("CART");
+        wordsSet.add("CAT");
+        HashSet prefixesSet = new HashSet();
+        prefixesSet.add("CAR");
+        prefixesSet.add("CARD");
+        prefixesSet.add("CART");
+        prefixesSet.add("CAT");
+        prefixesSet.add("CA");
+        prefixesSet.add("C");
+        prefixesSet.add("");
+        Dictionary dictionary = new Dictionary(prefixesSet,wordsSet);
+
+        WordSearcher wordSearcher = new WordSearcher();
+
+        System.out.print(wordSearcher.allWordsInGrid(dictionary,charsGrid));

```

Comment:
Tests should verify correctness of results automatically. Typically it is done using `assert` family of methods provided by JUnit's [`Assert` class](https://junit.org/junit4/javadoc/4.12/org/junit/Assert.html).
In this particular situation the best match would probably be combining `Assert.assertThat` and [`IsCollectionContaining.hasItems`](https://junit.org/junit4/javadoc/4.12/org/hamcrest/core/IsCollectionContaining.html#hasItems(T...))